internal/detect/detect.go

```diff
@@ -10,10 +11,12 @@ func Detect(logs scribe.Emitter) packit.DetectFunc {
 	return func(ctx packit.DetectContext) (packit.DetectResult, error) {
 		logs.Title("%s %s", ctx.Info.Name, ctx.Info.Version)
 
-		logs.Process("Checking working directory: %s", ctx.WorkingDir)
+		workingDir := api.GetWorkingDir(ctx.WorkingDir)
```

Comment:
The rest of this PR is just random clean-ups like dep updates, but this is the core: this ensures we always check the correct dir, otherwise, we check here and fail to detect hugo/mdBook, while the build step does it correctly.